Fix broken targeting with stacked accelerated Tile Entities

Shad0wlife opened the pull request:
Level.getEntitiesOfClass(Class, AABB) seems to also return entities in the 2 blocks below the given AABB for some reason. Excluding these by additionally matching the BlockPos of all found entities with the targeted BlockPos fixes the issue.

Fixes #22

AterAnimAvis commented:
Copying my reply from NeoForge's Discord just in case you missed it.

`Level.getEntitiesOfClass(Class, AABB) seems to also return entities in the 2 blocks`
This is because when registering the EntityType the mod never calls `EntityType$Builder#sized` so the bounding box of the entity will default to 1.8 blocks high.
This combined with the fact TIAB seems to be spawning the entity in the center of a block (+0.5) would result in you seeing the entities from 2 blocks below.

Shad0wlife commented:
Huh thanks. Then fixing that instead is probably the "correct" way to fix it. I'll look into adapting this PR

Shad0wlife committed:
Fix Entity Collision and Text Rendering

Fixes the entity collision to resolve the targeting issue where the 2 blocks above also targeted the lowest Block.

This made fixing the render locations of the text neccessary anyways, so the sometimes too dark text rendering was also fixed.

Shad0wlife commented:
Changed the fix method from the hacky blockpos filtering to properly resizing the entity and relocating the text rendering.

Also made the text render as FullBright because previously it needed backlight (a glowing block or transparent block) to actually be white for some reason.

RealMangorage commented:
So sorry for the late response.

I had no idea any activity was occurring here 🫨 

Anyways, ill pull in this change.

Do you know by chance which versions are affected?

Once I test these changes, I can pull in these changes 😄
